### PR TITLE
Update parameter binding data to provide ConnectionName

### LINKED ISSUE #1842: BlobTrigger fails to process a blob
This looks like a bug with Azure Functions Isolated Worker SDK.

### Observed issue

The setup is a blob-triggered function below.

```
[Function(nameof(BlobConcierge))]
public async Task Run([BlobTrigger("inspections/{blobName}", 
        Connection = "LossControl360:StorageConnectionString")] BlobInspection blobInspection,
        string blobName, FunctionContext context)
{
    var logger = context.GetLogger(nameof(BlobConcierge));

    // logic that's never invoked
}
```

When a new blob under the `inspections` container is dropped, the function throws.

`Error converting 1 input parameters for Function 'BlobConcierge': Cannot convert input parameter 'blobInspection' to type 'Tva.Endpoints.LossControl360.BlobInspection' from type 'Microsoft.Azure.Functions.Worker.Grpc.Messages.GrpcModelBindingData'. Error:System.InvalidOperationException: Blob storage connection configuration 'StorageConnectionString' does not exist. Make sure that it is a defined App Setting.`
<details><summary>Full stacktrace</summary>
   at Microsoft.Azure.Functions.Worker.BlobStorageBindingOptionsSetup.Configure(String name, BlobStorageBindingOptions options) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\Config\BlobStorageBindingOptionsSetup.cs:line 44
   at Microsoft.Extensions.Options.OptionsFactory`1.Create(String name)
   at Microsoft.Extensions.Options.OptionsMonitor`1.<>c__DisplayClass10_0.<Get>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location ---
   at System.Lazy`1.CreateValue()
   at Microsoft.Extensions.Options.OptionsCache`1.GetOrAdd(String name, Func`1 createOptions)
   at Microsoft.Extensions.Options.OptionsMonitor`1.Get(String name)
   at Microsoft.Azure.Functions.Worker.BlobStorageConverter.CreateBlobContainerClient(String connectionName, String containerName) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\BlobStorageConverter.cs:line 253
   at Microsoft.Azure.Functions.Worker.BlobStorageConverter.ConvertModelBindingDataAsync(Type targetType, BlobBindingData blobData) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\BlobStorageConverter.cs:line 118
   at Microsoft.Azure.Functions.Worker.BlobStorageConverter.ConvertFromBindingDataAsync(ConverterContext context, ModelBindingData modelBindingData) in D:\a\_work\1\s\extensions\Worker.Extensions.Storage.Blobs\src\BlobStorageConverter.cs:line 63 
</details>

The setting is found in the Functions App Configuration as `LossControl360__StorageConnectionString`.
image

To workaround this issue, I had to add a new `StorageConnectionString` setting to the Functions App Configuration. 

### Additional information

- .NET 6 / C#
- Relevant Nuget Packages
  - Azure.Storage.Blobs 12.17.0
  - Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs 6.0.1
  - Microsoft.Azure.Functions.Worker.Sdk 1.14.0
  - Microsoft.Azure.Functions.Worker 1.19.0
  - Microsoft.Azure.Functions.Worker.Extensions.Http 3.1.0
  - Microsoft.ApplicationInsights.WorkerService 2.21.0
  - Microsoft.Azure.Functions.Worker.ApplicationInsights 1.0.0-preview5

The failure started recently after updating NuGet packages.

### PR DESCRIPTION
**Pull request by liliankasem**
resolves https://github.com/Azure/azure-functions-dotnet-worker/issues/1842

There is a bug where the dotnet worker is not able to find and resolve a given connection name i.e. "MyApp:StorageConnection"; this is due to some preprocessing happening in the WebJobs extension around the connection name. This was a mistake in the initial implementation, we are supposed to be returning the full connection name and not the connection section key.

### CODE REVIEW

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs [hunk 1]
```diff
@@ -237,11 +237,10 @@ private ParameterBindingData ConvertBlobInputToParameterBindingData(BlobBaseClie
         private ParameterBindingData CreateParameterBindingData(string connection, string blobName, string containerName)
         {
             string connectionName = !string.IsNullOrEmpty(connection) ? _nameResolver.ResolveWholeString(connection) : string.Empty;
-            var connectionSection = _blobServiceClientProvider.GetWebJobsConnectionStringSection(connectionName);
 
             var blobDetails = new BlobParameterBindingDataContent()
             {
-                Connection = connectionSection.Key,
+                Connection = connectionName,
```
Comment (jviau) [inline sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs:243]: What are the impacts of this change? Will it work fine with older worker extensions?

Comment (liliankasem) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs:243]: Yup it should still work with older versions, the dotnet worker is doing the same work [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/extensions/Worker.Extensions.Storage.Blobs/src/Config/BlobStorageBindingOptionsSetup.cs); and there hasn't been an issue with plain connections like "MyConnection" - the more complex names like "MyApp:MyConnection" are broken today. Both scenarios should work with this change so I don't expect breaking changes. If there are any suggestions for more validation, I'm happy to do more testing/checks (so far I've tested various connection names with the local package and its working as expected)

**Review by fabiocav — APPROVED** [review]

**Comment by amnguye** [comment]
Looks like a lot of existing tests are failing.

BlobTrigger_BindingToParameterBindingData
Blob_IfBoundToParameterBindingData_Container_CreatesParameterBindingData
Blob_IfBoundToParameterBindingData_CreatesParameterBindingData
Blob_IfBoundToParameterBindingDataArray_CreatesParameterBindingDataArray
Blob_IfBoundToParameterBindingDataEnumerable_CreatesParameterBindingDataArray

Looks to be the same error:
```
Expected string length 19 but was 0. Strings differ at index 0.
Expected: "AzureWebJobsStorage"
But was:  <string.Empty>
```

Looks to be related to the Assert.AreEqual(ConnectionName, resultConnection);
Where it's expecting the value of "Connection" to have something.

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs (hunk 1 above)
Comment (liliankasem) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs:243]: This is however a breaking change for anyone that is ever binding to ParameterBindingData outside of the dotnet worker, so even though that should "never" happen, we should treat it as such.

**Comment by liliankasem** [comment]
@amnguye - addressed the test issue, mind giving this a second pass?

**Review by amnguye — APPROVED** [review]

**Pull request merged**